packages/flutter_tools/lib/src/isolated/devfs_web.dart

```diff
@@ -549,6 +549,9 @@ class WebAssetServer implements AssetReader {
     return modules;
   }
 
+  /// Whether to automatically detect which rendering backend to use.
+  bool autoDetect = false;
```

Comment:
If instead of a bool autodetect and a bool canvasKitRendering, you used a single enum, could you collapse the _resolveDartSdkJsFile methods into a map lookup?

Reply:
In the engine PR, we used two bools so I would like to keep it consistent here.

Reply:
The tooling code does not need to be consistent with the engine code.